src/cashier_frontend_new/src/modules/auth/signer/ii/IISignerAdapter.ts

```diff
@@ -0,0 +1,273 @@
+import { HttpAgent, type ActorSubclass, type Identity } from "@dfinity/agent";
+import { AuthClient } from "@dfinity/auth-client";
+import { BaseSignerAdapter } from "@windoge98/plug-n-play";
+import { type IIAdapterConfig, isIIAdapterConfig, Status } from "./type";
+import { Signer, type Transport } from "@slide-computer/signer";
+import { IITransport } from "./IITransport";
+import { FEATURE_FLAGS, HOST_ICP } from "$modules/shared/constants";
+import { getScreenDimensions } from "$modules/shared/utils";
+
+/**
+ * Account interface representing the connected user's account details.
+ */
+interface Account {
+  owner: string | null;
+  subaccount: string | null;
+}
+/**
+ * IISignerAdapter integrates Internet Identity (II) with the Plug and Play (PNP).
+ * By default PNP support a IIAdapter from plug-n-play, but it doesn't use signer-js
+ * which is required for the new transaction flow in Cashier.
```

Comment:
What is the "new transaction flow"? Is it ICRC-112? Since signer-js does not support either, can we migrate away from it?